src/groups/mqb/mqbauthn/mqbauthn_anonypassauthenticator.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright 2016-2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mqbauthn_anonypassauthenticator.h>
+
+// MQB
+#include <mqbplug_authenticator.h>
+
+// BMQ
+
+// BDE
+#include <bsl_iostream.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bsla_annotations.h>
+#include <bslma_allocator.h>
+#include <bslma_managedptr.h>
+#include <bsls_assert.h>
+
+namespace BloombergLP {
+namespace mqbauthn {
+
+// -----------------------------------
+// class AnonyPassAuthenticationResult
+// -----------------------------------
+
+AnonyPassAuthenticationResult::AnonyPassAuthenticationResult(
+    bsl::string_view   principal,
+    bsls::Types::Int64 lifetimeMs,
+    bslma::Allocator*  allocator)
+: d_principal(principal)
+, d_lifetimeMs(lifetimeMs)
+, d_allocator_p(allocator)
+{
+}
+
+AnonyPassAuthenticationResult::~AnonyPassAuthenticationResult()
+{
+}
+
+bsl::string_view AnonyPassAuthenticationResult::principal() const
+{
+    return d_principal;
+}
+
+const bsl::optional<bsls::Types::Int64>&
+AnonyPassAuthenticationResult::lifetimeMs() const
+{
+    return d_lifetimeMs;
+}
+
+// ----------------------------
+// class AnonyPassAuthenticator
+// ----------------------------
+
+AnonyPassAuthenticator::AnonyPassAuthenticator(
```

Comment:
`Anony` is not an abbreviation I've ever seen before.  Let's use the conventional `Anon`.